src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1640,6 +1642,97 @@ export const _handleCreateFormField: RequestHandler<
   )
 }
 
+/**
+ * NOTE: Exported for testing.
+ * Private handler for POST /forms/:formId/logic
+ * @precondition Must be preceded by request validation
+ * @security session
+ *
+ * @returns 200 with created logic object when successfully created
+ * @returns 403 when user does not have permissions to create logic
+ * @returns 404 when form cannot be found
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const _handleCreateLogic: RequestHandler<
+  { formId: string },
+  LogicDto | ErrorDto,
+  LogicDto
+> = (req, res) => {
+  const { formId } = req.params
+  const createdLogic = req.body
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  // Step 1: Retrieve currently logged in user.
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 3: Create form logic
+      .andThen((retrievedForm) =>
+        AdminFormService.createFormLogic(retrievedForm, createdLogic),
+      )
+      .map((createdLogic) => res.status(StatusCodes.OK).json(createdLogic))
+      .mapErr((error) => {
+        logger.error({
+          message: 'Error occurred when creating form logic',
+          meta: {
+            action: 'handleCreateLogic',
+            ...createReqMeta(req),
+            userId: sessionUserId,
+            formId,
+            createdLogic,
+          },
+          error,
+        })
+        const { errorMessage, statusCode } = mapRouteError(error)
+        return res.status(statusCode).json({ message: errorMessage })
+      })
+  )
+}
+
+/**
+ * Handler for POST /forms/:formId/logic
+ */
+export const handleCreateLogic = [
+  celebrate({
+    [Segments.BODY]: Joi.object({
+      logicType: Joi.string()
+        .valid(...Object.values(LogicType))
+        .required(),
+      conditions: Joi.array()
+        .items(
+          Joi.object({
+            field: Joi.string().required(),
+            state: Joi.string()
+              .valid(...Object.values(LogicConditionState))
+              .required(),
+            value: Joi.any().required(),
+            ifValueType: Joi.valid(...Object.values(LogicIfValue)),
+          }).unknown(true),
```

Comment:
maybe this is because of my lack of context but `ICondition` only seems to hold these 4 properties (`field/state/value/ifValueType`) so i'm unsure if the unknown is here only because of mongoose's `_id` property but if that's the case, maybe it might be better to add it as `_id: Joi.string().optional()`